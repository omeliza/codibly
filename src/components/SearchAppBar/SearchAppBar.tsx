import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import SearchIcon from '@mui/icons-material/Search';
import { search } from '../../store/slices/colorSlice';
import { useAppDispatch, useAppSelector } from '../../hooks/hooks';
import { Search, SearchIconWrapper, StyledInputBase } from './styles';

export default function SearchAppBar() {
  const value = useAppSelector(state => state.colorReducer.searchText);
  const dispatch = useAppDispatch();
  return (
    <Box sx={{ flexGrow: 1, mb: '20px' }}>
      <AppBar position="static">
        <Toolbar>
          <Typography
            variant="h6"
            noWrap
            component="div"
            sx={{ flexGrow: 1, display: { xs: 'none', sm: 'block' } }}
          >
            MUI
          </Typography>
          <Search>
            <SearchIconWrapper>
              <SearchIcon />
            </SearchIconWrapper>
            <StyledInputBase
              placeholder="Searchâ€¦"
              inputProps={{ 'aria-label': 'search' }}
              type="number"
              value={value}
              onChange={(e) => dispatch(search(e.target.value))}
            />
          </Search>
        </Toolbar>
      </AppBar>
    </Box>
  );
}