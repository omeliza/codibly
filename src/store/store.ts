import { combineReducers, configureStore } from "@reduxjs/toolkit";
// import { colorAPI } from "../services/DataService";
import colorReducer from './reducers/ColorSlice';

const rootReducer = combineReducers({
  colorReducer,
  // [colorAPI.reducerPath]: colorAPI.reducer
})

export const setupStore = () => {
  return configureStore({
    reducer: rootReducer,
    // middleware: (getDefaultMiddleware) =>
    //   getDefaultMiddleware().concat(colorAPI.middleware)
  })
}

export type RootState = ReturnType<typeof rootReducer>
export type AppStore = ReturnType<typeof setupStore>
export type AppDispatch = AppStore['dispatch']