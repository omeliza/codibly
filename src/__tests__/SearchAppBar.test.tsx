import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { setupStore } from '../store/store';

import SearchAppBar from './../components/SearchAppBar/SearchAppBar';

const store = setupStore();

describe('SearchAppBar', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
    <Provider store={store}>
      <SearchAppBar />
    </Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});