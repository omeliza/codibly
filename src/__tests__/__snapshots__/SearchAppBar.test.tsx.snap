// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAppBar should render correctly 1`] = `
<div
  className="MuiBox-root css-125p8oa"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-dudo9s-MuiTypography-root"
      >
        MUI
      </div>
      <div
        className="css-1orzy9s"
      >
        <div
          className="css-1iz4iln"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </div>
        <div
          className="MuiInputBase-root MuiInputBase-colorPrimary css-7mt4h1-MuiInputBase-root"
          onClick={[Function]}
        >
          <input
            aria-label="search"
            className="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Searchâ€¦"
            type="number"
          />
        </div>
      </div>
    </div>
  </header>
</div>
`;
